### PR TITLE
Add fmt command to qbec

### PR DESCRIPTION
**Pull request by harsimranmaan**
It is now possible to format jsonnet and yaml files using `qbec alpha fmt`.  To format only yaml files, use`qbec alpha fmt --yaml --jsonnet=false`. To format jsonnet files in-place use `qbec alpha fmt -w`.

This change also update go-jsonnet to v0.16.0 to use the new formatter capabilities. 
A change worth noting: `It is now allowed to have positional parameters after named ones`. For more details, please find the release notes at https://github.com/google/jsonnet/releases/tag/v0.16.0

### CODE REVIEW

File: internal/commands/fmt.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+}
+
+func isYamlFile(f os.FileInfo) bool {
+	name := f.Name()
+	return !f.IsDir() && !strings.HasPrefix(name, ".") && getFileType(name) == "yaml"
+}
+
+func isJsonnetFile(f os.FileInfo) bool {
+	name := f.Name()
+	return !f.IsDir() && !strings.HasPrefix(name, ".") && getFileType(name) == "jsonnet"
```
Comment (gotwarlost) [inline internal/commands/fmt.go:81]: `libsonnet` files?

File: internal/commands/fmt.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+}
+func isAFileOfInterestForFormatting(allowedExtension string, f os.FileInfo) bool {
```
Comment (gotwarlost) [inline internal/commands/fmt.go:83]: too wordy. maybe just `shouldFormat` or something?

File: internal/commands/fmt.go [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+		return formatJsonnet(in)
+	}
+	return nil, fmt.Errorf("unknown file type for file %q", filename)
+}
+
+const separator = "---\n"
+
+func formatYaml(in []byte) ([]byte, error) {
+	d := yaml.NewYAMLToJSONDecoder(bytes.NewReader(in))
+	var formatted []byte = make([]byte, 0)
```
Comment (gotwarlost) [inline internal/commands/fmt.go:214]: don't need the value to be assigned? `nil` should be ok right?

**Comment by gotwarlost** [comment]
Thanks for the PR! I still haven't looked at it in any detail. However, a couple of high-level observations:

* we need to make it very clear that YAML formatting will not preserve comments in the help text etc.
* should we introduce this as an `alpha fmt` command instead of a `fmt` command in case usage indicates that this needs backwards-incompatible flag changes  etc?

**Comment by harsimranmaan** [comment]
I have a working example that I am currently testing that would preserve comments but would not sort key. Alpha fmt seems like a good idea.

**Comment by gotwarlost** [comment]
I pulled down you branch and played with the `fmt` command to see how fast I could quickly construct a valid one :)

My personal opinion is that it is not intuitive to have the types be arguments and the files be options.

In other words, I'd prefer something like this
```
$ qbec alpha fmt  # processes both xsonnet and yaml files with an implicit argument of '.'
$ qbec alpha fmt --jsonnet ./foo ./bar # processes only xsonnet files in foo and bar
```

and perhaps we could have `-j` and `-y` shortcuts for the `--jsonnet` and `--yaml` options.
Thoughts? Specifically were you trying to maintain some command-line parity with existing formatters out there?

Don't make any changes until we discuss this.

**Comment by gotwarlost** [comment]
Pulled this down today for local testing. Loving it so far. In `-e` mode I think we should print the names of all the files that have the mismatch and not just error out on the first one.

So I did 
```
$ qbec alpha fmt --yaml -e
```

and it printed one file name with error. When I ran the `-w` version of the command, it ended up changing ~100 files. I found that unexpected.

Also should we have `-y` as a shortcut to `--yaml`?

**Comment by gotwarlost** [comment]
Also, it is not entirely clear to me if we should require a qbec root for what is a general purpose command. Thoughts? (don't make any changes yet)

**Comment by gotwarlost** [comment]
Some more observations:

* It took me awhile to figure out why this was a silent noop

```
$ qbec alpha fmt foobar.yaml # don't turn on yaml formatting but provided explicit name
```

* Also blank lines are not preserved in the yaml formatted output. Sometimes I use blank lines to delineate "sections" of some giant yaml (e.g. a ton of prometheus rules for example). I don't think this is a show-stopper and it's probably very difficult to fix anyway.

**Comment by gotwarlost** [comment]
Also, maybe JSON support at some point?
